resolver/resolver.cc

```diff
@@ -192,17 +192,19 @@ class ResolveConstantsWalk {
         }
         ast::Expression *resolvedScope = c->scope.get();
         if (auto *id = ast::cast_tree<ast::ConstantLit>(resolvedScope)) {
+            if (!id->symbol.exists()) {
+                id->symbol = resolveConstant(ctx, nesting, id->original);
```

Comment:
So far we've been intentionally trying to make `resolveConstant` non recursive.
Could we keep it lazy?